### ISSUE TITLE
Check if solution can be built against VS SDKs more recent than 2013

### ISSUE THREAD

**Issue by BrainCrumbz** [BrainCrumbz/NSpec.VsAdapter#1]
... e.g. like currently available VS 2015 SDK.

**Comment by sickboy** [comment]
@BrainCrumbz would be nice if the new SDK's are backwards compatible, that way you could support new VS/features, but still have the extension work with previous versions?

**Comment by BrainCrumbz** [comment]
Yep. Actually at the moment is more like SDK being *forward* compatible, but the concept still holds. 

E.g. currently this adapter is written and built against VS**2013** SDK. There are some new features that 2013 SDK introduces w.r.t to 2012, but as we choose 2013 as a base reference, we just take them for granted and take advantage of them, ignoring the 2012 scenario and basically being not compatible with it. 

If VS2015 SDK introduces some new features and adapter wants to use that, and still be compatible with VS2013, what we do is:

* keep on building against 2013 SDK, to remain compatible;
* then at runtime check (by reflection) if the new feature is present in the SDK linked at runtime in user environment, and in case use it

This is to keep working in 1 solution (or project, it depends), with the same code, and still being able to target more than one VS version. The alternative would be to have separate projects/solutions, built against different SDK versions, and hence separate code bases (or at least partially different), and this of course would complicate things, possibly more than the reflection-based approach.

**Comment by BrainCrumbz** [comment]
BTW, just to be fair, this issue is not about **using** the adapter, but about opening its source code and fiddling with it.
